rewrite-gradle/src/main/java/org/openrewrite/gradle/UpgradeDependencyVersion.java

```diff
@@ -124,7 +127,7 @@ public Validated<Object> validate() {
     @Value
     public static class DependencyVersionState {
         Map<String, GroupArtifact> versionPropNameToGA = new HashMap<>();
-        Map<GroupArtifact, String> gaToNewVersion = new HashMap<>();
+        Map<GroupArtifact, Object> gaToNewVersion = new HashMap<>();
```

Comment:
Perhaps good to document that this can contain a `GroupArtifact`, `String` and `Exception`, and how each are handled. Could otherwise be a bit puzzling to readers as it's not immediately apparent, even if it's necessary here.

Reply:
So this is keyed by `GroupArtifact`, but they value is either the `String` of the resolved version or the `MavenDownloadingException` representing an error in resolving the version to then be used to markup the dependency declaration later.

Reply:
```suggestion
        // The value is either a String representing the resolved version or a MavenDownloadingException representing an error during resolution.
        Map<GroupArtifact, Object> gaToNewVersion = new HashMap<>();
```

Seems good to you, @timtebeek?